src/App.tsx

```diff
@@ -1,61 +1,112 @@
+import React, { FormEventHandler, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setuserId] = useState('');
+  const [updatedTodos, setUpdatedTodos] = useState(todos);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+
+  const handleSubmit: FormEventHandler = (event) => {
+    event.preventDefault();
+
+    if (title && userId) {
+      const newTodo = {
+        id: Math.max(...updatedTodos.map(todo => todo.id)) + 1,
+        title,
+        completed: false,
+        userId: +userId,
+        user: getUser(+userId),
+      };
+
+      setUpdatedTodos([...updatedTodos, newTodo]);
+      setFormSubmitted(false);
+      setTitle('');
+      setuserId('');
```

Comment:
User ID should be set to null when form is cleared. Change setuserId('') to setuserId(null).